tests/dialects/test_pdl.py

```diff
@@ -94,6 +97,49 @@ def test_build_operation_replace():
         replace.verify()
 
 
+def test_build_pattern():
+    @Builder.implicit_region
+    def body() -> None:
+        root = pdl.OperationOp("test.op")
+        pdl.RewriteOp(root.results[0])
+
+    pattern = pdl.PatternOp(1, "pattern", body)
+
+    assert pattern.benefit == IntegerAttr.from_int_and_width(1, 16)
+    assert pattern.sym_name == StringAttr("pattern")
+    assert pattern.body is body
+
+    @irdl_op_definition
+    class DummyTerminator(IRDLOperation):
+        name = "dummy.terminator"
+        traits = frozenset([HasParent(pdl.PatternOp), IsTerminator()])
+
+    with pytest.raises(VerifyException) as verify_exception:
+
+        @Builder.implicit_region
+        def body() -> None:
+            DummyTerminator()
+
+        pattern = pdl.PatternOp(1, "pattern", body)
+        pattern.verify()
+    assert "expected body to terminate with a `pdl.rewrite`" in str(
+        verify_exception.value
+    )
+
+    with pytest.raises(VerifyException) as verify_exception:
+        root = pdl.OperationOp("test.op")
+
+        @Builder.implicit_region
+        def body() -> None:
+            pdl.RewriteOp(root.results[0])
+
+        pattern = pdl.PatternOp(1, "pattern", body)
+        pattern.verify()
+    assert "the pattern must contain at least one `pdl.operation`" in str(
+        verify_exception.value
+    )
```

Comment:
```suggestion
    with pytest.raises(VerifyException, match="the pattern must contain at least one `pdl.operation`"):
        root = pdl.OperationOp("test.op")

        @Builder.implicit_region
        def body() -> None:
            pdl.RewriteOp(root.results[0])

        pattern = pdl.PatternOp(1, "pattern", body)
        pattern.verify()
```

and above

Reply:
I wondered if there was a better way, thanks!